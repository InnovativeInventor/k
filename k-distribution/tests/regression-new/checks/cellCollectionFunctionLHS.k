module CELLCOLLECTIONFUNCTIONLHS
  imports BOOL
  imports INT
  imports MAP

  configuration <k> $PGM:K </k>
                <cells>
                  <cell multiplicity="*" type="Map">
                    <addr> 0 </addr>
                    <baz> 0 </baz>
                  </cell>
                </cells>


  syntax KItem ::= "foo"
  rule <k> foo => X </k> <cell> <addr> bar() </addr> <baz> X:Int </baz> </cell>
  rule <k> foo => 0 </k> <cell> <addr> 0 </addr> <baz> bar() </baz> </cell>

  syntax Int ::= bar() [function]
  rule bar() => 0
endmodule
