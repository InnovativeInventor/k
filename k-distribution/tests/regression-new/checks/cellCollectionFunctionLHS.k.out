[Error] Compiler: Illegal function symbol bar()_CELLCOLLECTIONFUNCTIONLHS_Int on LHS of rule. Consider adding `simplification` attribute to the rule if this is intended.
	Source(cellCollectionFunctionLHS.k)
	Location(17,56,17,61)
	17 |	  rule <k> foo => 0 </k> <cell> <addr> 0 </addr> <baz> bar() </baz> </cell>
	   .	                                                       ^~~~~
[Error] Compiler: Had 1 structural errors.
